File: trunk/api-cpp/rgma-client-check/InsertTuple.cpp
#include "rgma/RGMAException.h"
#include "rgma/TimeInterval.h"
#include "rgma/Storage.h"
#include "rgma/TimeUnit.h"
#include "rgma/PrimaryProducer.h"
#include "rgma/SupportedQueries.h"

#include <string>
#include <iostream>

using namespace glite::rgma;

int main(int argc, char* argv[]) {
  if (argc != 2)  {
    std::cerr<< "Usage: " << argv[0] << " <userId>" << std::endl;
    return 0;
  }

  try {

    TimeInterval terminationInterval(10, TimeUnit::MINUTES);
    PrimaryProducer pp (Storage::MEMORY, SupportedQueries::CH);

    std::string userTable("Default.userTable");
    std::string args(argv[1]);
    std::string predicate("WHERE userId = '" + args + "'");
    TimeInterval historyRP(10, TimeUnit::MINUTES);
    TimeInterval latestRP(10, TimeUnit::MINUTES);
    pp.declareTable(userTable,  predicate, historyRP, latestRP);

    std::string insert;
    insert.append("INSERT INTO Default.userTable (userId, aString, aReal, anInt)");
    insert.append(" VALUES ('");
    insert.append(args);
    insert.append("', 'C++ producer', 3.1415926, 42)");
    pp.insert(insert);
    pp.close();
  }

  catch (RGMAException rgmae) {
    std::cerr << "R-GMA application error in PrimaryProducer: " << rgmae.getMessage() << std::endl;
  }

  return 0;
}

File: api-java/rgma-client-check/InsertTuple.java
import org.glite.rgma.PrimaryProducer;
import org.glite.rgma.RGMAPermanentException;
import org.glite.rgma.RGMATemporaryException;
import org.glite.rgma.Storage;
import org.glite.rgma.SupportedQueries;
import org.glite.rgma.TimeInterval;
import org.glite.rgma.TimeUnit;

public class InsertTuple {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Usage: java InsertTuple <userId>");
            return;
        }
        try {
			PrimaryProducer pp = new PrimaryProducer(Storage.getMemoryStorage(), SupportedQueries.CH);
            String predicate = "WHERE userId = '" + args[0] + "'";
            TimeInterval historyRP = new TimeInterval(10, TimeUnit.MINUTES);
            TimeInterval latestRP = new TimeInterval(10, TimeUnit.MINUTES);
            pp.declareTable("Default.userTable", predicate, historyRP, latestRP);
            String insert = "INSERT INTO Default.userTable (userId, aString, aReal, anInt)" +
                   " VALUES ('" + args[0] + "', 'Java producer', 3.1415926, 42)";
            pp.insert(insert);
            pp.close();
        } catch (RGMATemporaryException e) {
            System.err.println("RGMATemporaryException: " + e.getMessage());
        } catch (RGMAPermanentException e) {
        	System.err.println("RGMAPermanentException: " + e.getMessage());

        }
    }
}
